test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,318 @@
+			Failf("Couldn't restore the original cluster size: %s", err)
+		}
+	})
+
+	It("should be able to delete nodes.", func() {
+		Logf("starting test %v", testContext.CloudConfig.NumNodes)
+
+		if !providerIs("gce") {
+			By(fmt.Sprintf("skipping ``should be able to delete nodes'' test, which is only supported for providers gce (not %s)",
+				testContext.Provider))
+			return
+		}
+
+		if testContext.CloudConfig.NumNodes < 2 {
+			By(fmt.Sprintf("skipping ``should be able to delete nodes'' test, which requires at lease 2 nodes (not %v)",
+				testContext.CloudConfig.NumNodes))
+			return
+		}
+
+		// Create a replication controller for a service that serves its hostname.
+		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+		name := "my-hostname-delete-node-" + string(util.NewUUID())
+		replicas := testContext.CloudConfig.NumNodes
+		createServeHostnameReplicationController(c, ns, name, replicas)
+		pods := waitForPodsCreated(c, ns, name, replicas)
+		waitForPodsRunning(c, pods)
+		verifyPodsListining(c, ns, name, pods)
+
+		By(fmt.Sprintf("decreasing cluster size to %v", replicas-1))
+		err := resizeNodeInstanceGroup(replicas - 1)
+		if err != nil {
+			Failf("Failed to decreas node instance group: %v", err)
+		}
+		err = waitForNodeInstanceGroupSize(replicas - 1)
+		if err != nil {
+			Failf("Failed to decreas node instance group: %v", err)
+		}
+		err = waitForClusterSize(c, replicas-1)
+		if err != nil {
+			Failf("Failed to decreas cluster size: %v", err)
+		}
+
+		By(fmt.Sprintf("verifying if the pods from the removed node are recreated"))
+		pods = waitForPodsCreated(c, ns, name, replicas)
+		for _, p := range pods.Items {
+			Logf("Pod: %v / %v", p.Name, p.Spec.Host)
+		}
+		waitForPodsRunning(c, pods)
+		verifyPodsListining(c, ns, name, pods)
+	})
+
+	It("should be able to add nodes.", func() {
+		Logf("starting test %v", testContext.CloudConfig.NumNodes)
+
+		if !providerIs("gce") {
+			By(fmt.Sprintf("Skipping ``should be able to delete nodes'' test, which is only supported for providers gce (not %s)",
+				testContext.Provider))
+			return
+		}
+
+		// Create a replication controller for a service that serves its hostname.
+		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+		name := "my-hostname-add-node-" + string(util.NewUUID())
+		createService(c, ns, name)
+		replicas := testContext.CloudConfig.NumNodes
+		createServeHostnameReplicationController(c, ns, name, replicas)
+		pods := waitForPodsCreated(c, ns, name, replicas)
+		waitForPodsRunning(c, pods)
+		verifyPodsListining(c, ns, name, pods)
+
+		By(fmt.Sprintf("increasing cluster size to %v", replicas+1))
+		err := resizeNodeInstanceGroup(replicas + 1)
+		if err != nil {
+			Failf("Failed to insrease node instance group: %v", err)
+		}
+		err = waitForNodeInstanceGroupSize(replicas + 1)
```

Comment:
Personally I like more whitespaces, but this doesn't matter. Please make it uniform across the file one way or another.